nixos/modules/services/web-apps/immich.nix

```
@@ -0,0 +1,575 @@
+        "immich-machine-learning.service"
+      ] ++ optional enableRedis "redis-immich.service";
+      wantedBy = ["multi-user.target"];
+
+      environment = serverEnv // {
+        SERVER_PORT = toString backendCfg.port;
+      }
+      // mapAttrs (_: toString) serverCfg.extraConfig
+      // mapAttrs (_: toString) backendCfg.extraConfig;
+
+      script = serverStartWrapper "${cfg.package}/bin/server";
+      serviceConfig = mkMerge [
+        (commonServiceConfig // serverServiceConfig)
+        {
+          EnvironmentFile = mkIf (backendCfg.environmentFile != null) backendCfg.environmentFile;
+        }
+      ];
+    };
+
+    systemd.services.immich-microservices = mkIf microservicesCfg.enable {
+      description = "Immich microservices (Self-hosted photo and video backup solution)";
+      after = [
+        "network.target"
+        "typesense.service"
+        "postgresql.service"
+        "immich-machine-learning.service"
+      ] ++ optional enableRedis "redis-immich.service";
+      wantedBy = ["multi-user.target"];
+
+      environment = serverEnv // {
+        MICROSERVICES_PORT = toString microservicesCfg.port;
+      }
+      // mapAttrs (_: toString) serverCfg.extraConfig
+      // mapAttrs (_: toString) microservicesCfg.extraConfig;
+
+      script = serverStartWrapper "${cfg.package}/bin/microservices";
+      serviceConfig = mkMerge [
+        (commonServiceConfig // serverServiceConfig)
+        {
+          EnvironmentFile = mkIf (microservicesCfg.environmentFile != null) microservicesCfg.environmentFile;
+        }
+      ];
+    };
+
+    systemd.services.immich-web = mkIf webCfg.enable {
+      description = "Immich web (Self-hosted photo and video backup solution)";
+      after = [
+        "network.target"
+        "immich-server.service"
+      ];
+      wantedBy = ["multi-user.target"];
+
+      environment = {
+        NODE_ENV = "production";
+        PORT = toString webCfg.port;
+        IMMICH_SERVER_URL = webCfg.serverUrl;
+        IMMICH_API_URL_EXTERNAL = webCfg.apiUrlExternal;
+      }
+      // mapAttrs (_: toString) webCfg.extraConfig;
+
+      script = ''
+        set -euo pipefail
+        export PUBLIC_IMMICH_SERVER_URL=$IMMICH_SERVER_URL
+        export PUBLIC_IMMICH_API_URL_EXTERNAL=$IMMICH_API_URL_EXTERNAL
+        exec ${cfg.package.web}/bin/web
+      '';
+      serviceConfig = commonServiceConfig // {
+        DynamicUser = true;
+        User = "immich-web";
+        Group = "immich-web";
+
+        MemoryDenyWriteExecute = false; # nodejs requires this.
+
+        TemporaryFileSystem = "/:ro";
+        BindReadOnlyPaths = [
+          "/nix/store"
+          "-/etc/resolv.conf"
+          "-/etc/nsswitch.conf"
+          "-/etc/hosts"
+          "-/etc/localtime"
+        ];
+      };
+    };
+
+    systemd.services.immich-machine-learning = mkIf mlCfg.enable {
+      description = "Immich machine learning (Self-hosted photo and video backup solution)";
+      after = ["network.target"];
```

Nit: Please add spaces around brackets:

```suggestion
      after = [ "network.target" ];
```